File: backend/app/schemas/priority.py
from typing import Optional, List
from pydantic import BaseModel

class PriorityItemSchema(BaseModel):
    id: str
    type: str
    priority_rank: int
    badge_label: str
    title: str
    lead_id: int
    lead_name: str
    lead_phone: str
    deal_value: float
    status: str
    scheduled_at: Optional[str] = None
    due_label: str
    reason: str
    recommended_action: str
    health_score: int
    health_category: str
    assigned_to_name: Optional[str] = None

class TodayPrioritiesResponse(BaseModel):
    total_priorities: int
    overdue_count: int
    today_actions_count: int
    post_visit_count: int
    hot_leads_count: int
    closing_count: int
    risk_count: int
    items: List[PriorityItemSchema]

File: frontend/src/types/priority.ts
export interface PriorityItem {
  id: string;
  type: 'overdue_followup' | 'today_site_visit' | 'today_followup' | 'post_visit' | 'hot_lead' | 'closing_opportunity' | 'lead_at_risk';
  priority_rank: number;
  badge_label: string;
  title: string;
  lead_id: number;
  lead_name: string;
  lead_phone: string;
  deal_value: number;
  status: string;
  scheduled_at?: string | null;
  due_label: string;
  reason: string;
  recommended_action: string;
  health_score: number;
  health_category: string;
  assigned_to_name?: string | null;
}

export interface TodayPrioritiesResponse {
  total_priorities: number;
  overdue_count: number;
  today_actions_count: number;
  post_visit_count: number;
  hot_leads_count: number;
  closing_count: number;
  risk_count: number;
  items: PriorityItem[];
}
